apps/cms.c

```diff
@@ -1063,12 +1063,12 @@ int cms_main(int argc, char **argv)
     } else if (operation & SMIME_SIGNERS) {
         int i;
         /*
-         * If detached data content we enable streaming if S/MIME output
-         * format.
+         * If detached data content and not signing pre-computed digest, we
+         * enable streaming if S/MIME output format.
          */
         if (operation == SMIME_SIGN) {
 
-            if (flags & CMS_DETACHED) {
+            if (flags & CMS_DETACHED && digestbin == NULL) {
```

Comment:
Please use `(flags & CMS_DETACHED) != 0 && digestbin == NULL` as the condition.

Reply:
Yeah, and this is one of the many items missing from the coding guidelines document, which is meanwhile at https://github.com/openssl/technical-policies/blob/master/policies/coding-style.md

Reply:
Will do. I couldn't find parenthesis around `(x & y)` and `(x | y)` in the coding style. Perhaps you should add it?

Reply:
There's a broken link on CONTRIBUTING.md to the coding style too. I opened #18015.

Reply:
Indeed the current guidelines say very little about parentheses.
The suggested changes on the coding guidelines at https://github.com/openssl/openssl/issues/10725 add this:

> Avoid needless parentheses, e.g., replace `if ((ptr == NULL) && (!f(x)))` by `if (ptr == NULL && !f(x))`.

Yet in the LHS conjunct of `(flags & CMS_DETACHED) != 0 && digestbin == NULL`
the parameters are needed simply because of C operator precedence: 
for whatever reason, `&` binds less tight than `!=`
See, e.g., https://en.cppreference.com/w/c/language/operator_precedence

Reply:
Sure, but I had `if (flags & CMS_DETACHED && digestbin == NULL)` first, which doesn't need parentheses for precedence. To check flags with != and == is another style point to add.

Reply:
> There's a broken link on CONTRIBUTING.md to the coding style too. I opened #18015.

Great you found this and proposed a correction.
Using search engines, I had not found the new location
https://www.openssl.org/policies/technical/coding-style.html